app/src/controllers/note.ts

```diff
@@ -104,28 +140,60 @@ const controller = {
     }
   },
 
-  async listNotes(req: Request<{ activityId: string }>, res: Response, next: NextFunction) {
+  /**
+   * @function listNoteHistory
+   * Get a list of all note histories for the given activityId
+   */
+  async listNoteHistory(req: Request<{ activityId: string }>, res: Response, next: NextFunction) {
     try {
-      const response = await noteService.listNotes(req.params.activityId);
-      res.status(200).json(response);
+      const response = await noteHistoryService.listNoteHistory(req.params.activityId);
+
+      // Only return notes flagged as shown when called by proponent
+      if (req.currentAuthorization?.attributes.includes('scope:self')) {
+        const filtered = response.filter((x) => x.shownToProponent);
+        return res.status(200).json(filtered);
+      }
+
+      return res.status(200).json(response);
     } catch (e: unknown) {
       next(e);
     }
   },
 
-  async updateNote(req: Request<never, never, Note>, res: Response, next: NextFunction) {
+  /**
+   * @function addNote
+   * Updates a note history
+   * Adds a new note to an existing history if one was given
+   */
+  async updateNoteHistory(
+    req: Request<{ noteHistoryId: string }, never, NoteHistory & { note: string | undefined }>,
+    res: Response,
+    next: NextFunction
+  ) {
     try {
-      const response = await noteService.updateNote({
-        ...req.body,
-        ...generateCreateStamps(req.currentContext),
+      await noteHistoryService.updateNoteHistory({
+        activityId: req.body.activityId,
+        bringForwardDate: req.body.bringForwardDate,
+        bringForwardState: req.body.bringForwardState,
+        escalateToSupervisor: req.body.escalateToSupervisor,
+        escalateToDirector: req.body.escalateToDirector,
+        noteHistoryId: req.params.noteHistoryId,
+        shownToProponent: req.body.shownToProponent,
+        title: req.body.title,
+        type: req.body.type,
+        isDeleted: false,
         ...generateUpdateStamps(req.currentContext)
```

Comment:
You use straight function call here but use `const stamps = ...` in the above create function